charts/rstudio-connect/files/job.tpl

```diff
@@ -267,20 +267,42 @@ spec:
               {{- end }}
             {{- end }}
           {{- end }}
-          {{- if any (ne (len $requests) 0) (ne (len $limits) 0) }}
+          {{- if or (ne (len $requests) 0) (ne (len $limits) 0) }}
           resources:
             {{- if ne (len $requests) 0 }}
             requests:
-              {{- range $key, $val := $requests }}
-              {{ $key }}: {{ toYaml $val }}
+              {{- if $requests.cpu }}
+              cpu: {{ $requests.cpu }}
+              {{- else }}
+              cpu: 500m
+              {{- end }}
+              {{- if $requests.memory }}
+              memory: {{ $requests.memory }}
+              {{- else }}
+              memory: 1Gi
               {{- end }}
             {{- end }}
             {{- if ne (len $limits) 0 }}
             limits:
-              {{- range $key, $val := $limits }}
-              {{ $key }}: {{ toYaml $val }}
+              {{- if $limits.cpu }}
+              cpu: {{ $limits.cpu }}
+              {{- else }}
+              cpu: 500m
+              {{- end }}
+              {{- if $limits.memory }}
+              memory: {{ $limits.memory }}
+              {{- else }}
+              memory: 1Gi
               {{- end }}
             {{- end }}
+          {{- else }}
+          resources:
+            requests:
+              cpu: 500m
+              memory: 1Gi
+            limits:
+              cpu: 500m
+              memory: 1Gi
```

Comment:
I think my preference would be for these defaults to be defined in the `values.yaml` rather than here in the template. We'll want to add a new config section under `launcher.templateValues.pod.resources` so that the configured values are available inside of `$templateData.pod.resources`, then we can merge those values with the values provided on the `.Job.resourceLimits` - this would be more consistent with how the other job overrides work.

Take `VolumeMounts` for example:

```yaml
{{- if or (ne (len .Job.volumes) 0) (ne (len $templateData.pod.volumeMounts) 0) }}
          volumeMounts:
            {{- range .Job.volumeMounts }}
            - {{ nindent 14 (toYaml .) | trim -}}
            {{- end }}
            {{- range $templateData.pod.volumeMounts }}
            - {{ nindent 14 (toYaml .) | trim -}}
            {{- end }}
{{- end }}
```

Except in this case we'll want to prefer values from the `.Job` and fall back to values in `$templateData` rather than appending.